include/irr/asset/material_compiler/IR.h

```diff
@@ -346,14 +346,15 @@ class IR : public core::IReferenceCounted
             ET_MICROFACET_DIFFUSE,
             ET_MICROFACET_SPECULAR,
             //ET_SHEEN,
-            ET_COATING
+            ET_COATING,
+            ET_DIELECTRIC
```

Comment:
lets talk about consistency.

Either we differentiate between rough and smooth (microfacet and delta) or we dont and allow 0 roughness materials.

So either you have
DELTA_SPECULAR + MICROFACET_SPECULAR
DELTA_COATING + MICROFACET_COATING
DELTA_DIELECTRIC + MICROFACET_DIELECTRIC
DELTA_THIN_DIELECTRIC + MICROFACET_THIN_DIELECTRIC

where a MICROFACET_THIN_DIELECTRIC is a TODO (backends assert that its not used)

Or just
MICROFACET_SPECULAR
MICROFACET_COATING
MICROFACET_DIELECTRIC
MICROFACET_THIN_DIELECTRIC

where a rough thindielectric is a TODO (backends assert that roughness is 0)

Reply:
oh, btw you need a `DELTA_TRANSMISSION` (to implement opacity mask)

Reply:
> oh, btw you need a `DELTA_TRANSMISSION` (to implement opacity mask)

yep, im aware, just not the time yet